src/App.jsx

```diff
@@ -1,14 +1,65 @@
 import './App.css';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
+import DATA from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const [entryData, setEntryData] = useState(DATA);
+  const [localColor, setLocalColor] = useState('green');
+  const [remoteColor, setRemoteColor] = useState('blue');
+
+  const LOCAL_SENDER = entryData[0].sender;
+  const REMOTE_SENDER = entryData[1].sender;
+
+  const handleColorChange = (sender, color) => {
+    if (sender === LOCAL_SENDER){
+      setLocalColor(color);
+    } else if (sender === REMOTE_SENDER) {
+      setRemoteColor(color);
+    }
+  };
+
+  const updateEntryLikedState = (entryId) => {
+    setEntryData(entries => {
```

Comment:
Nice use of the callback setter style. In this application, it doesn't really matter whether we use the callback style or the value style, but it's good practice to get in the habit of using the callback style.